### PR TITLE
Do not try to remove stale label on PRs

### PR DESCRIPTION
**Pull request by yubiuser**
# PR Master branch

**What does this PR aim to accomplish?:**

PR https://github.com/pi-hole/pi-hole/pull/5252 added the workflow to remove the 'stale' label from issues immediately when a stale issue received new comments. However, the github event `issue_comment` also triggers on  **PR comments** (see https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issue_comment).

As we don't  handle stale PR with this workflow PR write permission is not granted and the workflow will fail (see https://github.com/pi-hole/pi-hole/actions/runs/5109474838/jobs/9184306155)

This PR adds an additional check to verify the action is not triggered by  comments on pull requests. (see https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issue_comment-on-issues-only-or-pull-requests-only) 

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

### CODE REVIEW

**Comment by yubiuser** [comment]
Second commit should fix

Screenshot at 2023-05-31 21-58-01

**Review by DL6ER — DISMISSED** [review]

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -39,7 +42,6 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3.5.2
       - name: Remove 'stale' label
-        run: gh issue edit ${{ github.event.issue.number }} --remove-label $stale_label
-        env:
```
Comment (PromoFaux) [inline .github/workflows/stale.yml:43]: You dropped the `env:`. I'll push a fix at the same time as tweaking the conditional..

Comment (yubiuser) [reply .github/workflows/stale.yml:43]: Thanks. This was not planned.

Comment (yubiuser) [reply .github/workflows/stale.yml:43]: Looks good. Let's try this.

**Review by PromoFaux — DISMISSED** [review]

**Review by PromoFaux — APPROVED** [review]

**Comment by PromoFaux** [comment]
This was merged into other repos, not sure how I missed this one

**Pull request merged**